sdks/python/apache_beam/runners/worker/opcounters.py

```diff
@@ -24,11 +24,99 @@
 
 
+class TransformIoCounter(object):
+  """Class to track time and bytes consumed while reading from IO.
+
+  Examples of IO can be side inputs, shuffle, or streaming state.
+  """
+
+  def check_step(self):
+    pass
+
+  def __exit__(self, exc_type, exc_value, traceback):
+    pass
+
+  def __enter__(self):
+    pass
+
+  def add_bytes_read(self, n):
+    pass
+
+
+class SideInputReadCounter(TransformIoCounter):
+  """Tracks time and bytes consumed while reading from side inputs.
+
+  This class is designed to track consumption of side inputs across fused steps.
+  We represent a side input as a declaring step, and an input index, which are
+  represented as an io_target.
+  The declaring step is the step that originally receives the side input for
+  consumption, and the input index in which the declaring step receives the side
+  input that we want to identify.
+  Note that the declaring step originally receives the side input, but it may
+  not be the only step that spends time reading from this side input.
+  """
+
+  def __init__(self, counter_factory, state_sampler, declaring_step,
+               input_index):
+    """Create a side input read counter.
+
+    Args:
+      counter_factory: A counters.CounterFactory to create byte counters.
+      state_sampler: A statesampler.StateSampler to transition into read states.
+      declaring_step: The step that directly receives the side input initially.
+      input_index: The index of the side input in the list of inputs of the
+        declaring step.
+
+    The side input is uniquely identified by (declaring_step, input_index);
+    where declaring_step is the step that receives the PCollectionView as a
+    side input, and input_index is the index of the PCView within the list of
+    inputs.
+    """
+    self._counter_factory = counter_factory
+    self._state_sampler = state_sampler
+    self.declaring_step = declaring_step
+    self.input_index = input_index
+    self.check_step()
+
+  def check_step(self):
+    """Update the current running step.
+
+    Due to the fusion optimization, user code may choose to emit the data
+    structure that holds side inputs (Iterable, Dict, or others). This call
+    updates the current step, to attribute the data consumption to the step
+    that is responsible for it.
+    """
+    current_state = self._state_sampler.current_state()
+    operation_name = current_state.name.step_name
+    self.scoped_state = self._state_sampler.scoped_state(
+        self.declaring_step,
+        'read-sideinput',
+        io_target=counters.side_input_id(operation_name, self.input_index))
+    self.bytes_read_counter = self._counter_factory.get_counter(
+        CounterName(
+            'read-sideinput-byte-count',
+            step_name=self.declaring_step,
+            io_target=counters.side_input_id(operation_name, self.input_index)),
+        Counter.SUM)
+
+  def add_bytes_read(self, n):
```

Comment:
<!--new_thread; commit:1e64e1f8b880a9fe4450c84cf1c7c5694eb3567e; resolved:0-->
Let's change `n` to `count`.